rio-code/src/main/java/frc/robot/subsystems/RobotMode.java

```diff
@@ -0,0 +1,200 @@
+        coralReefAlignPose = new CoralReefAlignPose();
+        coralReefAligned = new CoralReefAligned();
+        coralScorePose = new CoralScorePose();
+    
+        ejectCoralPose = new EjectCoralPose();
+    }
+
+    @Override
+    public void periodic() {
+
+        Logger.log("current mode", currentMode.getName());
+
+        if (driveMode != null) {
+            switch (driveMode) {
+                case TeleopDrive:
+    
+                    drivetrain.setControl(teleopDrive.withVelocityX(SwerveXSupplier.get() < 0 ? Math.pow(SwerveXSupplier.get() * MaxSpeed, 2) : -Math.pow(SwerveXSupplier.get() * MaxSpeed, 2))
+                                                .withVelocityY(SwerveYSupplier.get() < 0 ? Math.pow(SwerveYSupplier.get() * MaxSpeed, 2) : -Math.pow(SwerveYSupplier.get() * MaxSpeed, 2))
+                                                .withRotationalRate(-SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+    
+                    break;
+    
+                case FieldCentric:
+    
+                    drivetrain.setControl(fieldCentric.withVelocityX(SwerveXSupplier.get() * MaxSpeed)
+                                                .withVelocityY(SwerveYSupplier.get() * MaxSpeed)
+                                                .withRotationalRate(SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+                    
+                    break;
+            
+                case RobotCentric:
+    
+                    drivetrain.setControl(robotCentric.withVelocityX(SwerveXSupplier.get() * MaxSpeed)
+                                                    .withVelocityY(SwerveYSupplier.get() * MaxSpeed)
+                                                    .withRotationalRate(SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+    
+                    break;
+                
+                case Brake:
+    
+                    break;
+    
+                case Auton:
+    
+                    break;
+    
+                default:
+                    break;
+            }
+        }
+        
+    }
+
+    public void setDriveModeCommand(Command newDriveMode) {
+        if (currentDriveMode != null) {
+            currentDriveMode.cancel();
+        }
+        currentDriveMode = newDriveMode;
+        currentDriveMode.schedule();
+    }
+
+    public void setCurrentMode(Command newMode) {
+        if (currentMode != null) {
+            currentMode.cancel();
+        }
+        currentMode = newMode;
+        currentMode.schedule();
+    }
+
+    public void setDriveMode(DriveMode driveMode) {
+        this.driveMode = driveMode;
+    }
+
+    public void setSwerveControl(Supplier<Double> swerveXSupplier, Supplier<Double> swerveYSupplier, Supplier<Double> swerveRotationSupplier) {
+        this.SwerveXSupplier = swerveXSupplier;
+        this.SwerveYSupplier = swerveYSupplier;
+        this.SwerveRotationSupplier = swerveRotationSupplier;
```

Comment:
This also seems like it should be combined with the drive mode stuff above. Let's try to consolidate to a single function instead of 3.